src/Microsoft.ML.StandardTrainers/Standard/MulticlassClassification/MulticlassNaiveBayesTrainer.cs

```diff
@@ -13,20 +13,20 @@
 using Microsoft.ML.Runtime;
 using Microsoft.ML.Trainers;
 
-[assembly: LoadableClass(NaiveBayesTrainer.Summary, typeof(NaiveBayesTrainer), typeof(NaiveBayesTrainer.Options),
+[assembly: LoadableClass(NaiveBayesMulticlassTrainer.Summary, typeof(NaiveBayesMulticlassTrainer), typeof(NaiveBayesMulticlassTrainer.Options),
     new[] { typeof(SignatureMulticlassClassifierTrainer), typeof(SignatureTrainer) },
-    NaiveBayesTrainer.UserName,
-    NaiveBayesTrainer.LoadName,
-    NaiveBayesTrainer.ShortName, DocName = "trainer/NaiveBayes.md")]
+    NaiveBayesMulticlassTrainer.UserName,
+    NaiveBayesMulticlassTrainer.LoadName,
+    NaiveBayesMulticlassTrainer.ShortName, DocName = "trainer/NaiveBayes.md")]
 
-[assembly: LoadableClass(typeof(MulticlassNaiveBayesModelParameters), null, typeof(SignatureLoadModel),
-    "Multi Class Naive Bayes predictor", MulticlassNaiveBayesModelParameters.LoaderSignature)]
+[assembly: LoadableClass(typeof(NaiveBayesMulticlassModelParameters), null, typeof(SignatureLoadModel),
+    "Multi Class Naive Bayes predictor", NaiveBayesMulticlassModelParameters.LoaderSignature)]
 
-[assembly: LoadableClass(typeof(void), typeof(NaiveBayesTrainer), null, typeof(SignatureEntryPointModule), NaiveBayesTrainer.LoadName)]
+[assembly: LoadableClass(typeof(void), typeof(NaiveBayesMulticlassTrainer), null, typeof(SignatureEntryPointModule), NaiveBayesMulticlassTrainer.LoadName)]
 
 namespace Microsoft.ML.Trainers
 {
-    public sealed class NaiveBayesTrainer : TrainerEstimatorBase<MulticlassPredictionTransformer<MulticlassNaiveBayesModelParameters>, MulticlassNaiveBayesModelParameters>
+    public sealed class NaiveBayesMulticlassTrainer : TrainerEstimatorBase<MulticlassPredictionTransformer<NaiveBayesMulticlassModelParameters>, NaiveBayesMulticlassModelParameters>
```

Comment:
`NaiveBayesMulticlassTrainer` => `NaiveBayesMulticlassClassificationTrainer`? Why are we dropping `Classification` from these names?

Reply:
as noted in the bug description, it seems for ModelParameters  we are consistent in *not* using the word "Classification"  . I am following the same convention in this PR.

Example:
- LinearBinaryModelParameters
- FastTreeBinaryModelParameters

I don't think we have solved the "Classification" debate completely yet #2623 

Here is my observation :
- For Trainers, we use "Classification" 
- For ModelParameters, we do not use "Classification"

---
In reply to: [266627123](https://github.com/dotnet/machinelearning/pull/2968#discussion_r266627123) [](ancestors = 266627123)

Reply:
> For Trainers, we use "Classification"
For ModelParameters, we do not use "Classification"

Why is this a valid policy?

Reply:
I cannot claim its a "valid" policy,  but one observation I have is that  

- we do not use "Classification" anywhere for ModelParameters
  - CalibratedModelParametersBase
  - LinearBinaryModelParameters 
- in some cases it leads to long names  e.g. `FastForestMulticlassClassificationModelParameter`

We have 2 options:

A. Follow the naming convention used by `Trainers` i.e.  "<>BinaryClassification" and "MulticlassClassification" uniformly.   
B.  Not  use "Classification"  for ModelParameters .

Which one you favor ?  @eerhardt @sfilipi